java/data-management/strings-i/finding-a-substring-in-a-string.md

```diff
@@ -35,3 +39,20 @@ int indexFirst = s.indexOf("dog");
 int indexLast = s.lastIndexOf("dog");
 //indexLast: 15
 ```
+
+---
+
+## Revision
+
+Would what be the result of the following code?
+
+```java
+String s = "Check out this cat, my cat, the best cat really";
+int indexLast = s.indexOf("cat");
```

Comment:
Why `indexLast` name here? It's making this kind of a trick question :)

Reply:
int indexLast = s.lastIndexOf("dog");
//indexLast: 15 

We have this example, in the insight so the question is following the same format.

Too tricky?

Reply:
Well the code in the question is using `indexOf`, not `lastIndexOf` (although the name implies last index)

Reply:
I think changing the variable's name from `indexLast` to `index` should do the trick

Reply:
Yup. Or use `lastIndexOf` instead of `indexOf`